tests/testutils/testcase.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/testutil"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+type Testcase struct {
```

Comment:
Please document exported struct. Also, does it need to be exported?

Reply:
Will do.  I'd say so since some proposed cases use the builder and its receiver methods (e.g. https://github.com/signalfx/splunk-otel-collector/pull/145/files#diff-10deb04b733fce760ce0b30c42f5862339f0e2d7aced0e2f8a934e637eca3adb) and I believe golint discourages this.